### ISSUE TITLE
Gradients: can't handle non-existing gradients with gateGradients = true

### ISSUE THREAD

**Issue by csaladin94** [eaplatanios/tensorflow_scala#44]
Hi there!

I found a bug when using `tf.Gradients.gradients(..., gateGradients=true)` on a graph that uses a `tf.concatenate` operation.
I found the bug because I wanted to use the AdaGrad optimizer.

A simple example to reproduce the issue:

```
import org.platanios.tensorflow.api._

object GatedGradientBugReproducer {
  def main(args: Array[String]): Unit = {
    val a = tf.variable("a", FLOAT32, Shape(1))
    val concat = tf.concatenate(Seq(a, Tensor(1.0f)))
    val loss = concat.sum()

    tf.Gradients.gradients(Seq(loss), Seq(a), gateGradients = true) // Throws NullPointerException
  }
}

```

I dug a bit in the code, the exception is thrown when tf wants to create a `ControlFlow.tuple ` with the inputs containing a `null` value.

Thanks a lot for having a look!

Best,
Carlo

**Comment by eaplatanios** [comment]
@csaladin94 Thanks for pointing this out! I think this has already been fixed in the master branch although I haven't released new artifacts on Sonatype for it. I will release new artifacts tomorrow and then your code should probably run fine. :)

**Comment by csaladin94** [comment]
@eaplatanios Thanks for the fast answer, glad to hear that!
I'll come back to you as soon as I have tried my code with the update.

**Comment by csaladin94** [comment]
@eaplatanios Just compiled the project myself from the master branch, can confirm that it the bug is fixed.

**Comment by eaplatanios** [comment]
@csaladin94 Great, thanks for checking it! :)
